enclave-manager/web/src/client/KurtosisClientContext.tsx

```diff
@@ -0,0 +1,88 @@
+import { createContext, PropsWithChildren, useContext, useEffect, useState } from "react";
+import { KurtosisClient } from "./KurtosisClient";
+import { Alert, AlertDescription, AlertIcon, AlertTitle, Flex, Heading, Spinner } from "@chakra-ui/react";
+import { assertDefined, isDefined, isStringTrue, stringifyError } from "../utils";
+import { AuthenticatedKurtosisClient } from "./AuthenticatedKurtosisClient";
+import { LocalKurtosisClient } from "./LocalKurtosisClient";
+
+type KurtosisClientContextState = {
+  client: KurtosisClient | null;
+};
+
+const KurtosisClientContext = createContext<KurtosisClientContextState>({ client: null });
+
+export const KurtosisClientProvider = ({ children }: PropsWithChildren) => {
```

Comment:
For this provider we can split the jwtToken handling out but keep the code related to requestedApiHost, preloadedPackage

Reply:
Do you mean to encapsulate the jwt effect specifically? Something like:

```
type KurtosisJWTDetectorProps = {
  onJWTDetected: (jwt: string) => void;
};

export const KurtosisJWTDetector = ({ onJWTDetected }: KurtosisJWTDetectorProps) => {
  useEffect(() => {
    const receiveMessage = (event: MessageEvent) => {
      const message = event.data.message;
      switch (message) {
        case "jwtToken":
          const value = event.data.value;
          if (isDefined(value)) {
            onJWTDetected(value);
          }
          break;
      }
    };
    window.addEventListener("message", receiveMessage);
    return () => window.removeEventListener("message", receiveMessage);
  }, []);
  return <></>;
};
```

In my opinion this is excessive encapsulation - and `KurtosisClientProvider` would still have a state variable for storing the jwt token.

Reply:
I think we can chat about these topics all together (referencing earlier comment) so I better understand your reasoning in this design.